### PR TITLE
Stop using loader functions for swap page

### LINKED ISSUE #1243: Stop using loader functions
We use loaders to obtain initial data on some pages, both in minifront and in prax.

It's difficult to manage ordered/dependent execution of loaders, and loaders block page execution, which has lead to several issues, particularly in minifront where connection success is not guaranteed.

Currently, we call a function at the beginning of minifront loaders that will throw if the loader is expected to time out. This is not a good pattern.

We should examine using either a sustained zustand slice that keeps necessary data ready, or, using connect-query as intended by connectrpc.

https://connectrpc.com/docs/web/query/getting-started/

### PR DESCRIPTION
**Pull request by jessepinho**
This PR removes the loader function from the swap route, and instead uses ZQuery, resulting in instant route changes when a user navigates to `/swap`.

Closes #1243

### CODE REVIEW

File: apps/minifront/src/components/send/send-form/index.tsx
Comment (jessepinho) [inline apps/minifront/src/components/send/send-form/index.tsx:1]: Not strictly related to this PR, just something I noticed while working on it

File: apps/minifront/src/components/swap/swap-loader.ts [hunk 1]
```diff
@@ -1,42 +0,0 @@
...
-export type SwapLoaderResponse = UnclaimedSwapsWithMetadata[];
-
-const getAndSetDefaultAssetBalances = async (swappableAssets: Metadata[]) => {
-  const balancesResponses = await getSwappableBalancesResponses();
-  const stakingTokenAssetMetadata = await getStakingTokenMetadata();
-
-  // set initial denom in if there is an available balance
-  if (balancesResponses[0]) {
-    useStore.getState().swap.setAssetIn(balancesResponses[0]);
-    useStore.getState().swap.setAssetOut(swappableAssets[0]!);
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-loader.ts:25]: These two `set*` calls are now made in the ZQuery object's `fetch` function.

File: apps/minifront/src/components/swap/swap-loader.ts [hunk 2]
```diff
@@ -1,42 +0,0 @@
...
-    useStore.getState().swap.setStakingAssetMetadata(stakingTokenAssetMetadata);
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-loader.ts:26]: This is now accessed via `useStakingTokenMetadata`

File: apps/minifront/src/components/swap/swap-loader.ts [hunk 3]
```diff
@@ -1,42 +0,0 @@
...
-};
-
-export const SwapLoader: LoaderFunction = async (): Promise<null> => {
-  await abortLoader();
-  const assets = await getAllAssets();
-  const swappableAssets = assets.filter(isSwappable);
-
-  const balancesResponses = await getAndSetDefaultAssetBalances(swappableAssets);
-  useStore.getState().swap.setBalancesResponses(balancesResponses);
-  useStore.getState().swap.setSwappableAssets(swappableAssets);
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-loader.ts:39]: These are now ZQuery objects

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx [hunk 4]
```diff
@@ -143,21 +138,41 @@ export const TokenSwapInput = () => {
...
-
-          <div className='flex flex-col gap-2 pt-2'>
-            <ArrowRight size={16} className='text-muted-foreground' />
-          </div>
-
-          <div className='flex h-full flex-col gap-2'>
-            <AssetSelector assets={swappableAssets} value={assetOut} onChange={setAssetOut} />
-            {assetOut && <BalanceValueView valueView={assetOutBalance} />}
-          </div>
+          <FadeIn condition={!!balancesResponses.data && !!swappableAssets.data}>
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/token-swap-input.tsx:141]: This fades in the input/output selectors once the necessary data has loaded:

screencap

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx [hunk 5]
```diff
@@ -76,24 +79,16 @@ const tokenSwapInputSelector = (state: AllSlices) => ({
...
-    hasStakingTokenMeta,
-  } = useStoreShallow(tokenSwapInputSelector);
+  const stakingTokenMetadata = useStakingTokenMetadata();
+  const balancesResponses = useBalancesResponses();
+  const swappableAssets = useSwappableAssets();
+  const { amount, setAmount, assetIn, setAssetIn, assetOut, setAssetOut, priceHistory } =
+    useStoreShallow(tokenSwapInputSelector);
   // State to manage privacy warning display
-  const [showNonNativeFeeWarning, setshowNonNativeFeeWarning] = useState(false);
+  const [showNonNativeFeeWarning, setShowNonNativeFeeWarning] = useState(false);
```
Comment (jessepinho) [inline apps/minifront/src/components/swap/swap-form/token-swap-input.tsx:88]: typo fix

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx (hunk 4 above)
Comment (grod220) [reply apps/minifront/src/components/swap/swap-form/token-swap-input.tsx:141]: Can we handle the error case as well? Maybe some red text with a loading error message?

**Review by grod220 — APPROVED** [review]
LGTM

File: apps/minifront/src/components/swap/swap-form/token-swap-input.tsx (hunk 4 above)
Comment (jessepinho) [reply apps/minifront/src/components/swap/swap-form/token-swap-input.tsx:148]: done

**Comment by jessepinho** [comment]
@TalDerei 
> @jessepinho can you fold in this [comment](https://github.com/penumbra-zone/web/pull/1268#discussion_r1643130786) to reduce code duplication and have the _warning_ use our shiny new zquery loaders?

Ah, sure — but I'm going to do that as a separate super-small PR, since as part of that, I want to calculate `showNonNativeFeeWarning` via a selector too, and that's out of scope for this PR.

**Pull request merged**